frontend/src/components/CopyMenuItem.tsx

```diff
@@ -29,6 +31,9 @@ export const CopyMenuItem: React.FC<CopyMenuItemProps> = ({
       onClick={async () => {
         await awaitCopy?.()
         clipboard.copy(value)
+        if (onCopied) {
+          setTimeout(onCopied, 900)
```

Comment:
@evanrbowers Can we just move this logic into the useClipboard hook? You can add an onCopied callback to that so we keep the timing logic in the hook please. Nice UX improvement!